# Convert DNA Files 
As a way to introduce how Exceptions work and how they can be created, this program takes DNA sequences and converts them into two different formats: FASTA and FASTQ. The program itself not only converts these DNA sequences but also throws user-made execptions should the files are not able to be converted.  
## About the Assignment
Although the program is made to show how FASTA and FASTQ files can be converted, the main purpose of the assignment is to show how exceptions can be created and used to ensure the program runs well. The Exception, RecordFormatException, comes from the class of the same name; the class extends to the Exception class. The exception is thrown when the sequence or defline is not in the correct format it needs to be in. Other Exceptions, like IOExceptions, are also used. Other classes and interface used in this assignment are used to either read, write, record different files as well as convert them into another format if needed.  
#### About FASTA and FASTQ
Defintion and explanation given by [Stack Excange](https://bioinformatics.stackexchange.com/questions/14/what-is-the-difference-between-fasta-fastq-and-sam-file-formats?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)
* In bioinformatics, FATA and FASTQ are text-based formats that store sequence data and sequence metadata. 
  - FASTA stores a variable number of sequence records, and for each record it stores the sequence itself, and a sequence ID. Each record starts with a header line whose first character is >, followed by the sequence ID. The next lines of a record contain the actual sequence.In the context of nucleotide sequences, FASTA is mostly used to store reference data; that is, data extracted from a curated database; the above is adapted from GtRNAdb (a database of tRNA sequences).
  - FASTQ record contains a sequence of quality scores for each nucleotide. FASTQ was conceived to solve a specific problem of FASTA files: when sequencing, the confidence in a given base call (that is, the identity of a nucleotide) varies. This is expressed in the Phred quality score. FASTA had no standardised way of encoding this. FASTQ files are mostly used to store short-read data from high-throughput sequencing experiments. As a consequence, the sequence and quality scores are usually put into a single line each, and indeed many tools assume that each record in a FASTQ file is exactly four lines long, even though this isnâ€™t guaranteed.
